dotnet/src/Functions/Functions.OpenApi/Model/RestApiOperation.cs

```diff
@@ -215,6 +205,24 @@ internal string BuildQueryString(IDictionary<string, object?> arguments)
         return string.Join("&", segments);
     }
 
+    /// <summary>
+    /// Makes the current instance unmodifiable.
+    /// </summary>
+    internal void Freeze()
+    {
```

Comment:
Consider an `isFrozen` flag so we can exit early if the instance is already frozen.

Reply:
Fixed.